modules/behavior/draw_way.js

```diff
@@ -412,6 +413,81 @@ export function behaviorDrawWay(context, wayID, mode, startGraph) {
         });
     };
 
+    /** see PR #8671 */
+    function followMode() {
+        if (_didResolveTempEdit) return;
+
+        try {
+
+            // get the last 2 added nodes.
+            // check if they are both part of only oneway (the same one)
+            // check if the ways that they're part of are the same way
+            // find index of the last two nodes, to determine the direction to travel around the existing way
+            // add the next node to the way we are drawing
+
+            // if we're drawing an area, the first node = last node.
+            const isDrawingArea = _origWay.nodes[0] === _origWay.nodes.slice(-1)[0];
+
+            const [secondLastNodeId, lastNodeId] = _origWay.nodes.slice(isDrawingArea ? -3 : -2);
+
+            if (!lastNodeId || !secondLastNodeId) {
+                context.ui().flash
+                    .duration(4000)
+                    .iconName('#iD-icon-no')
+                    .label(t('operations.follow.error.needs_more_initial_nodes'))();
+                return;
+            }
+
+            const lastNodesParents = startGraph.parentWays(startGraph.entity(lastNodeId));
```

Comment:
Apologies if being picky here, but I think we could improve the 'error' handling here and better inform user which actions should he take to achieve the follow behavior.
Currently if any of the last 2 nodes is not following the pre-existing path, the exception will happen in startGraph.entity() function and will be caught as unknown_error. I think it's better we handle these cases with error.needs_more_initial_nodes info.
Additionally, there is an edge case where node is pre-existing but it's not vertex (has no parent way).

To sum it up I propose the following:
Throw needs_more_initial_nodes if:
- <2 nodes have been drawn
- Any of the last 2 nodes are not matching pre-existing node (startGraph.entity would throw error, though I propose .hasEntity be used for this)
- Any of the last 2 nodes has 0 parents

Reply:
good points, I've made this change and tested it quite a bit